File: CSharpSample_DLL/P2PSample/TransferCallForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace P2PSample
{
    public partial class TransferCallForm : Form
    {
        public TransferCallForm()
        {
            InitializeComponent();
        }

        private string TransferNumber = string.Empty;

        public string GetTransferNumber()
        {
            return TransferNumber;
        }

        public int GetReplaceLineNum()
        {
            if (TextBoxLineNum.Text.Length <= 0)
            {
                return 0;
            }

            return int.Parse(TextBoxLineNum.Text);
        }

        private void TransferCallForm_Load(object sender, EventArgs e)
        {
            TransferNumber = "";
            TextBoxTranferNumber.Text = "";

            TextBoxLineNum.Text = "";
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (TextBoxTranferNumber.Text.Length > 0)
            {
                TransferNumber = TextBoxTranferNumber.Text;
                TextBoxTranferNumber.Text = "";
                this.DialogResult = DialogResult.OK;
            }
            else
            {
                MessageBox.Show("Please enter the transfer number.", "Information", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
            }
        }

        private void TransferCallForm_FormClosing(object sender, FormClosingEventArgs e)
        {

        }


    }
}

File: VBSample/SIPSample/TransferCallForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Public Partial Class TransferCallForm
	Inherits Form
	Public Sub New()
		InitializeComponent()
	End Sub

	Private TransferNumber As String = String.Empty

	Public Function GetTransferNumber() As String
		Return TransferNumber
	End Function

	Public Function GetReplaceLineNum() As Integer
		If TextBoxLineNum.Text.Length <= 0 Then
			Return 0
		End If

		Return Integer.Parse(TextBoxLineNum.Text)
	End Function

	Private Sub TransferCallForm_Load(sender As Object, e As EventArgs)
		TransferNumber = ""
		TextBoxTranferNumber.Text = ""

		TextBoxLineNum.Text = ""
	End Sub

	Private Sub Button1_Click(sender As Object, e As EventArgs)
		If TextBoxTranferNumber.Text.Length > 0 Then
			TransferNumber = TextBoxTranferNumber.Text
			TextBoxTranferNumber.Text = ""
			Me.DialogResult = DialogResult.OK
		Else
			MessageBox.Show("Please enter the transfer number.", "Information", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
		End If
	End Sub
End Class
